app/controllers/api/v2/hosts_controller.rb

```diff
@@ -123,7 +123,32 @@ def destroy
       eos
 
       def status
-        render :json => { :status => @host.host_status }.to_json if @host
+        Foreman::Deprecation.api_deprecation_warning('The /status route is deprecated, please use the new /status/configuration instead')
+        render :json => { :status => @host.get_status(HostStatus::ConfigurationStatus).to_label }.to_json if @host
+      end
+
+      api :GET, "/hosts/:id/status/:type", "Get status of host"
+      param :id, :identifier_dottable, :required => true
+      param :type, [ HostStatus::Global ] + HostStatus.status_registry.to_a.map { |s| s.status_name.underscore}, :required => true, :desc => <<-eos
+status type, can be one of configuration
+* global
+* configuration
+* build
+      eos
+      # TRANSLATORS: API documentation - do not translate
+      description 'Returns string representing a host status of a given type'
+      def get_status
+        case params[:type]
```

Comment:
How will plugins get their status types recognised here?  Shouldn't it be looked up in the registry?